adaptive/src/main/java/com/google/accompanist/adaptive/FoldAwareColumn.kt

```diff
@@ -0,0 +1,321 @@
+            height,
+            arrangementSpacing.roundToPx()
+        )
+
+        override fun IntrinsicMeasureScope.maxIntrinsicHeight(
+            measurables: List<IntrinsicMeasurable>,
+            width: Int
+        ) = MaxIntrinsicHeightMeasureBlock(orientation)(
+            measurables,
+            width,
+            arrangementSpacing.roundToPx()
+        )
+    }
+}
+
+/**
+ * Inherits from [RowColumnMeasurementHelper] to place children in a fold-aware manner
+ */
+private class FoldAwareColumnMeasurementHelper(
+    orientation: LayoutOrientation,
+    arrangement: (Int, IntArray, LayoutDirection, Density, IntArray) -> Unit,
+    arrangementSpacing: Dp,
+    crossAxisSize: SizeMode,
+    crossAxisAlignment: CrossAxisAlignment,
+    measurables: List<Measurable>,
+    placeables: Array<Placeable?>
+) : RowColumnMeasurementHelper(
+    orientation,
+    arrangement,
+    arrangementSpacing,
+    crossAxisSize,
+    crossAxisAlignment,
+    measurables,
+    placeables
+) {
+    /**
+     * Copy of [placeHelper] that has been modified for FoldAwareColumn implementation
+     */
+    @OptIn(ExperimentalComposeUiApi::class)
+    fun foldAwarePlaceHelper(
+        placeableScope: Placeable.PlacementScope,
+        measureResult: RowColumnMeasureHelperResult,
+        crossAxisOffset: Int,
+        layoutDirection: LayoutDirection,
+        foldBoundsPx: Rect?
+    ) {
+        with(placeableScope) {
+            val layoutBounds = coordinates!!.boundsInWindow()
+
+            var placeableY = 0
+
+            for (i in measureResult.startIndex until measureResult.endIndex) {
+                val placeable = placeables[i]
+                placeable!!
+                val mainAxisPositions = measureResult.mainAxisPositions
+                val crossAxisPosition = getCrossAxisPosition(
+                    placeable,
+                    (measurables[i].parentData as? RowColumnParentData),
+                    measureResult.crossAxisSize,
+                    layoutDirection,
+                    measureResult.beforeCrossAxisAlignmentLine
+                ) + crossAxisOffset
+                if (orientation == LayoutOrientation.Horizontal) {
+                    placeable.place(
+                        mainAxisPositions[i - measureResult.startIndex],
+                        crossAxisPosition
+                    )
+                } else {
+                    val relativeBounds = Rect(
+                        left = 0f,
+                        top = placeableY.toFloat(),
+                        right = placeable.width.toFloat(),
+                        bottom = (placeableY + placeable.height).toFloat()
+                    )
+                    val absoluteBounds =
+                        relativeBounds.translate(layoutBounds.left, layoutBounds.top)
+
+                    // If placeable overlaps fold, push placeable below
+                    if (foldBoundsPx?.let { absoluteBounds.overlaps(it) } == true &&
```

Comment:
When testing, found that `overlaps` will return false if a placeable has a width of zero (ex: `Spacer` with only height specified) - do we want to change the logic to just check if the bounds overlap vertically, or do we want to keep the current logic?

Reply:
Agreed - just changed the logic to check only if the top/bottom ranges of the Rects overlap.

This changed some of the logic for calculating the bounds of a Rect in the current window, so let me know if the updates still look ok!